Cargo.toml

```diff
@@ -51,21 +48,32 @@ minimal = []
 compiler = ["miniscript/compiler"]
 verify = ["bitcoinconsensus"]
 default = ["key-value-db", "electrum"]
-electrum = ["electrum-client"]
-esplora = ["reqwest", "futures"]
 compact_filters = ["rocksdb", "socks", "lazy_static", "cc"]
 key-value-db = ["sled"]
-async-interface = ["async-trait"]
 all-keys = ["keys-bip39"]
 keys-bip39 = ["tiny-bip39", "zeroize"]
 rpc = ["bitcoincore-rpc"]
 
+# We currently provide mulitple implementations of `Blockchain`, all are
+# blocking except for the `EsploraBlockchain` which can be either async or
+# blocking, depending on the HTTP client in use.
+#
+# - Users wanting asynchronous HTTP calls should enable `async-interface` to get
+#   access to the asynchronous method implementations. Then, if Esplora is wanted,
+#   enable `esplora` AND `reqwest`.
+# - Users wanting blocking HTTP calls can use any of the other blockchain
+#   implementations (`compact_filters`, `electrum`, or `esplora`). Users wanting to
+#   use Esplora should enable `esplora` AND `ureq`.
+async-interface = ["async-trait"]
+electrum = ["electrum-client"]
+esplora = ["futures"]           # Requires one of: `ureq` or `reqwest`.
```

Comment:
makes more sense that `futures` is turned on by `async-interface` rather than `esplora` now.

Reply:
oh yes, you are right in that this doesn't make sense. How about if `futures` is turned on by `reqwest` however because that's the only thing that uses them. (Commented without coding.)

Reply:
I've had another think about how the features are combined and made an effort to make usage more ergonomic. Its in a separate patch to ease review commit: `c01f515 Improve feature combinations for ureq/reqwest`. From the commit log of that patch:
```
    ...

    With this patch applied we can do

    - `cargo check --no-default-features --features=use-esplora-reqwest`
    - `cargo check --no-default-features --features=use-esplora-ureq`
    - `cargo check --features=use-esplora-ureq`
    - `cargo check --no-default-features --features=async-trait`
```

Reply:
Includes update to CI

Reply:
Good idea. This looks a better to me. I'd prefer to remove the `use-` from the front of these though.